smack-experimental/src/main/java/org/jivesoftware/smackx/httpfileupload/element/SlotRequest.java

```diff
@@ -0,0 +1,71 @@
+/**
+ *
+ * Copyright © 2017 Grigory Fedorov
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smackx.httpfileupload.element;
+
+import org.jivesoftware.smack.packet.IQ;
+
+/**
+ * Upload slot request.
+
+ * @author Grigory Fedorov
+ * @see <a href="http://xmpp.org/extensions/xep-0363.html">XEP-0363: HTTP File Upload</a>
+ */
+public class SlotRequest extends IQ {
+    public static final String ELEMENT = "request";
+    public static final String NAMESPACE = "urn:xmpp:http:upload";
+
+    private final String filename;
+    private final long size;
+    private final String contentType;
+
+    public SlotRequest(String filename, long size, String contentType) {
+        super(ELEMENT, NAMESPACE);
+
+        this.filename = filename;
+        this.size = size;
```

Comment:
This should be sanitized, i.e. throw IllegalArgumentException when negative or zero.